revision-history.yaml

```diff
@@ -8,6 +8,7 @@ revisionHistory:
     - Add connect, invalidate to syntax highlighting
     - Add alternative `regreset` syntax
     - Add literal identifiers to allow for legal numeric fields
+    - Initial ABI description.
```

Comment:
Shouldn't this go into a different revision history yml file...?

Reply:
Is this a change log for the repository or just for the spec?  Are they versioned together?  The ABI has more than passive ties to the constructs described in the spec.

Adding enums, say, will require some ABI specification since they can appear on ports.

I was taking the strategy of co-versioning the documents, but having the ABI versions be orthogonal to the spec version.  That way when enums are added, they can be specified for both the type-lowering abi (v1) and the aggregate preservation abi (v2).

Reply:
I don't know the answer yet to whether the spec version and the repo version should match, today they are identical but we could have branches for the api docs too but that starts getting crazy...? 

The way it's set up now works pretty well to make point releases of the spec on branches of the repo. Not sure how branchy/different the abi doc would be from the spec. We could always make branches specific for the abi doc point versions if we find ourselves needing it

Reply:
The contents of this are used to populate the Version History section of the spec. If the spec and the ABI are co-versioned, then there's not much issue other than we should add two sub-sections here and have the version history pulled out differently, e.g.,:

```yaml
thisVersion:
  spec:
    - change legal identifiers to be emoji only
  abi:
    - redstone ABI defined
```

Reply:
@seldridge how would that work if we go to an actual version? Like let's say we release 1.1.0 of ABI doc and version 3.1.0 of spec doc, then start adding more to `thisVersion` for both... what does that look like?

To me if we want different docs we just have different versions of this file with their own version history parts below `thisVersion`...?

Then i think the release of this repo should just follow the spec version or if we really care we could stick spec and ABI version into the release tag

Reply:
The question is if the spec and ABI documents are under the same version. If they are, then this will work. If they are separate, then a separate file would be better. Lenharth is arguing that they need the same version because the spec is going to add stuff which the ABI needs to talk about. We are expecting to have multiple ABIs which is why there are v1 and v2 of the ABI internal to the document.

Mechanically, every version will have a `spec` and `abi` keys whose values are a list of changes.

Reply:
ahhh ok, i misread this as having different versions of the ABI document. I agree if they are all versioned the same as a document with different versions of the ABI in each document, that is much simpler